hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -81,33 +81,73 @@ default String getName() {
     @Nonnull
     CompletableFuture<Void> getFuture();
 
-    /**
-     * Attempts to cancel execution of this job. The job will be completed
-     * after the job has been stopped on all the nodes.
-     *
-     * Starting from version 0.6, <code>job.getFuture().cancel()</code> fails
-     * with an exception.
-     */
-    boolean cancel();
-
     /**
      * Waits for the job to complete and throws exception if job is completed
-     * with an error.
+     * with an error. Does not return if the job is suspended. Never returns
+     * for streaming (unbounded) jobs unless they fail.
      *
-     * Shorthand for <code>job.getFuture().get()</code>
+     * <p>Shorthand for <code>job.getFuture().get()</code>.
      */
     default void join() {
         Util.uncheckRun(() -> getFuture().get());
     }
 
     /**
-     * Cancels the current execution if the job is currently running and
-     * schedules a new execution with the current member list of the Jet cluster
+     * Gracefully stops the current execution and schedules a new execution
+     * with the current member list of the Jet cluster. Can be called to
+     * manually make use of added members.
      *
-     * @throws IllegalStateException if the job has been already completed
+     * <p>Conceptually it's equivalent to {@link #suspend()} & {@link
+     * #resume()}.
      *
-     * @return true if the current execution of the job is cancelled
+     * <p>You cannot restart a suspended job, to do that, call {@link
+     * #resume()}.
+     *
+     * @throws IllegalStateException if the job is not running, for example it
+     * has been already completed, is not yet running, is already restarting,
+     * suspended etc.
      */
-    boolean restart();
+    void restart();
 
+    /**
+     * Gracefully suspend the current execution. Job status will become {@link
+     * JobStatus#SUSPENDED}. To resume the job, call {@link #resume()}.
+     *
+     * <p>If the job does not do {@linkplain JobConfig#setProcessingGuarantee
+     * state snapshots}, it will be suspended anyway. When resumed, it will
+     * start with an empty state.
+     *
+     * <p>This call returns quickly and a suspension process is initiated. This
```

Comment:
maybe we can explicitly say: `This call returns quickly and a suspension process is initiated in the background`